cadquery/occ_impl/importers/assembly.py

```diff
@@ -0,0 +1,198 @@
+    """
+    Import a step file into an assembly.
+
+    :param assy: An Assembly object that will be packed with the contents of the STEP file.
+    :param path: Path and filename to the STEP file to read.
+
+    :return: None
+    """
+
+    # Document that the step file will be read into
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+
+    # Create and configure a STEP reader
+    step_reader = STEPCAFControl_Reader()
+    step_reader.SetColorMode(True)
+    step_reader.SetNameMode(True)
+    step_reader.SetLayerMode(True)
+    step_reader.SetSHUOMode(True)
+
+    # Read the STEP file
+    status = step_reader.ReadFile(path)
+    if status != IFSelect_RetDone:
+        raise ValueError(f"Error reading STEP file: {path}")
+
+    # Transfer the contents of the STEP file to the document
+    step_reader.Transfer(doc)
+
+    # Shape and color tools for extracting XCAF data
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+    layer_tool = XCAFDoc_DocumentTool.LayerTool_s(doc.Main())
+
+    def _process_simple_shape(label, parent_location=None, parent_name=None):
+        shape = shape_tool.GetShape_s(label)
+
+        # Tracks the RGB color value and whether or not it was found
+        cq_color = None
+
+        # Load the name of the part in the assembly, if it is present
+        name = None
+        if parent_name is not None and parent_name != assy.name:
+            name = parent_name
+        else:
+            name_attr = TDataStd_Name()
+            label.FindAttribute(TDataStd_Name.GetID_s(), name_attr)
+            name = str(name_attr.Get().ToExtString())
+
+        # Process the color for the shape, which could be of different types
+        color = Quantity_Color()
+        cq_color = cq.Color(0.50, 0.50, 0.50)
+        if color_tool.GetColor_s(label, XCAFDoc_ColorSurf, color):
+            r = color.Red()
+            g = color.Green()
+            b = color.Blue()
+            cq_color = cq.Color(r, g, b)
+
+        # Handle the location if it was passed down form a parent component
+        if parent_location is not None:
+            assy.add(
+                cq.Shape.cast(shape),
+                name=name,
+                color=cq_color,
+                loc=cq.Location(parent_location),
+            )
+        else:
+            assy.add(cq.Shape.cast(shape), name=name, color=cq_color)
+
+        # Check all the attributes of all the children to find the subshapes and any names
+        for j in range(label.NbChildren()):
+            child_label = label.FindChild(j + 1)
+            attr_iterator = TDF_AttributeIterator(child_label)
+            while attr_iterator.More():
+                current_attr = attr_iterator.Value()
+
+                # Get the type name of the attribute so that we can decide how to handle it
+                if current_attr.DynamicType().Name() == "TNaming_NamedShape":
```

Comment:
Could you add an explanation/comment regarding NamedShape and the other thing below (GraphNode).